app/src/main/java/fr/free/nrw/commons/media/MediaClient.java

```diff
@@ -161,4 +169,147 @@ public MediaClient(MediaInterface mediaInterface) {
+                                Timber.e(filename);
+                                Map.Entry<String, Caption> captionEntry = labels.entrySet().iterator().next();
+                                Caption caption = captionEntry.getValue();
+                                return caption.getValue();
+
+                        } catch (Exception e) {
+                            return "No caption";
+                        }
+                    }
+                        return "No caption";
+
+                })
+                .singleOrError();
+    }
+
+    /**
+     * Fetches Structured data from API
+     *
+     * @param filename
+     * @return a map containing caption and depictions (empty string in the map if no caption/depictions)
+     */
+
+    public Single<JsonObject> getCaptionAndDepictions(String filename)  {
+        return mediaDetailInterface.fetchStructuredDataByFilename(Locale.getDefault().getLanguage(), filename)
+                .map(mediaDetailResponse -> {
+                        return fetchCaptionandDepictionsFromMediaDetailResponse(mediaDetailResponse);
+                })
+                .singleOrError();
+    }
+
+    /**
+     * Parses the mediaDetailResponse from API to extract captions and depictions
+     * @param mediaDetailResponse Response obtained from API for Media Details
+     * @return a map containing caption and depictions (empty string in the map if no caption/depictions)
+     */
+
+    @SuppressLint("CheckResult")
+    private JsonObject fetchCaptionandDepictionsFromMediaDetailResponse(MediaDetailResponse mediaDetailResponse) {
+        JsonObject mediaDetails = new JsonObject();
+        if (mediaDetailResponse != null && mediaDetailResponse.getSuccess() != null && mediaDetailResponse.getSuccess() == 1 && mediaDetailResponse.getEntities() != null) {
+            Map<String, CommonsWikibaseItem> entities = mediaDetailResponse.getEntities();
+            try {
+                Map.Entry<String, CommonsWikibaseItem> entry = entities.entrySet().iterator().next();
+                CommonsWikibaseItem commonsWikibaseItem = entry.getValue();
+                try {
+                    Map<String, Caption> labels = commonsWikibaseItem.getLabels();
+                    Map.Entry<String, Caption> captionEntry = labels.entrySet().iterator().next();
+                    Caption caption = captionEntry.getValue();
+                    JsonElement jsonElement = new JsonPrimitive(caption.getValue());
+                    mediaDetails.add("Caption", jsonElement);
+                } catch (Exception e) {
+                    JsonElement jsonElement = new JsonPrimitive("No caption");
+                    mediaDetails.add("Caption", jsonElement);
+                }
+
+                try {
+                    LinkedTreeMap statements = (LinkedTreeMap) commonsWikibaseItem.getStatements();
+                    ArrayList<LinkedTreeMap> listP245962 = (ArrayList<LinkedTreeMap>) statements.get("P180");
```

Comment:
Please rename the variable to something like `depictionsList`, since P245962 has no meaning on the production server.

Reply:
renamed to `p180ItemList` in the next commit

Reply:
I think `depictionsList` is better, since P180 has no meaning on the beta server, or on any potential future server.

Reply:
It is a list of depictions, not of Wikibase items, right?